File: Cap 12/12.7 - Poliformismo/12.7.3 - Poliformismo Ad-hoc de Sobrecarga/Exemplo 1/polimadcsob1.cpp
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacão de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 12 - Topico 12.7
 * 
 * Programa ..: POLIFORMISMO AD-HOC DE SOBRECARGA
 * 
 */

#include <iostream>
using namespace std;

class CLS_AREA {
public:
  int AREA(int X) {
    return X * X;
  }

  double AREA(double R, double H) {
    return R * R * 3.14159 * H;
  }

  int AREA(int X, int Y, int Z) {
    return X * Y * Z;
  }
};

int main(void) {
	
  CLS_AREA RESPOSTA;

  cout << "POLIFORMISMO DE SOBRECARGA" << endl;
  cout << endl;
  cout << "Area: Quadrado ..: " << RESPOSTA.AREA(5) << endl;
  cout << "Area: Cubo ......: " << RESPOSTA.AREA(5, 6, 7) << endl;
  cout << "Area: Cilindro ..: " << RESPOSTA.AREA(7, 3) << endl;

  return 0;
}

File: Cap 12/12.7 - Poliformismo/12.7.3 - Poliformismo Ad-hoc de Sobrecarga/Exemplo 1/polimadcsob1.kt
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacão de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 12 - Topico 12.7
 * 
 * Programa ..: POLIFORMISMO AD-HOC DE SOBRECARGA
 * 
 */

import kotlin.math.PI

class CLS_AREA {
  
    fun AREA(X: Int): Int {
        return X * X
    }

    fun AREA(R: Double, H: Double): Double {
        return R * R * PI * H
    }

    fun AREA(X: Int, Y: Int, Z: Int): Int {
        return X * Y * Z
    }
    
}

fun main() {
  
    var RESPOSTA = CLS_AREA()

    println("POLIFORMISMO DE SOBRECARGA")
    println()
    println("Area: Quadrado ..: ${RESPOSTA.AREA(5)}")
    println("Area: Cubo ......: ${RESPOSTA.AREA(5, 6, 7)}")
    println("Area: Cilindro ..: ${RESPOSTA.AREA(7.0, 3.0)}")
    
}
